### PR TITLE
fix(tabs): focus on keyboard navigation

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

- Fixed focus on keyboard navigation, await tabindex update before applying `.focus()` on selected tab.
- Fixed incorrect placement of tabindex on the container of tab and not the button

### Todo:
 - [x] Add spec test that checks for correct focus on keyboard navigation

## Testing Instructions

See [deploy preview demo](https://deploy-preview-2301--patternfly-elements.netlify.app/components/tabs/demo/).

## Reviewer Notes

Check for accessibility of keyboard navigation and correctness of set focus.

### CODE REVIEW

File: elements/pfe-tabs/BaseTabs.ts [hunk 1]
```diff
@@ -298,9 +296,11 @@ export abstract class BaseTabs extends LitElement {
     }
   }
 
-  #select(selectedTab: BaseTab): void {
+  async #select(selectedTab: BaseTab): Promise<void> {
     this.#activate(selectedTab);
     this.#focusTab = selectedTab;
+    await this.updateComplete;
+    this.#focusTab.focus();
```
Comment (bennypowers) [inline elements/pfe-tabs/BaseTabs.ts:303]: LGTM. Reserving this comment for @nikkimk 's review

**Review by bennypowers — APPROVED** [review]

**Review by nikkimk — APPROVED** [review]
Worked for me.

File: elements/pfe-tabs/BaseTabs.ts (hunk 1 above)
Comment (nikkimk) [reply elements/pfe-tabs/BaseTabs.ts:303]: Looks okay to me.

File: elements/pfe-tabs/BaseTab.ts [hunk 2]
```diff
@@ -67,10 +69,10 @@ export abstract class BaseTab extends LitElement {
 
   #activeChanged() {
     if (this.active && !this.disabled) {
-      this.removeAttribute('tabindex');
+      this._button.removeAttribute('tabindex');
```
Comment (bennypowers) [inline elements/pfe-tabs/BaseTab.ts:72]: Could we just do this is the lit template? Imperatively modifying DOM that lit controls is fragile

Comment (bennypowers) [reply elements/pfe-tabs/BaseTab.ts:72]: PTAL @zeroedin

Comment (zeroedin) [reply elements/pfe-tabs/BaseTab.ts:72]: Fixed in https://github.com/patternfly/patternfly-elements/pull/2310/commits/9ac7e3d8fe6c8a2fc81e16c9c9e51695cad4b067

**Pull request merged**